consensus/consensus_v2.go

```diff
@@ -67,32 +43,32 @@ func (consensus *Consensus) handleMessageUpdate(payload []byte) {
 	// Handle validator intended messages first
 	case t == msg_pb.MessageType_ANNOUNCE &&
 		intendedForValidator &&
-		consensus.validatorSanityChecks(msg):
+		consensus.validatorSanityChecks(msg, senderKey):
 		consensus.onAnnounce(msg)
 	case t == msg_pb.MessageType_PREPARED &&
 		intendedForValidator &&
-		consensus.validatorSanityChecks(msg):
+		consensus.validatorSanityChecks(msg, senderKey):
 		consensus.onPrepared(msg)
 	case t == msg_pb.MessageType_COMMITTED &&
 		intendedForValidator &&
-		consensus.validatorSanityChecks(msg):
+		consensus.validatorSanityChecks(msg, senderKey):
 		consensus.onCommitted(msg)
 	// Handle leader intended messages now
 	case t == msg_pb.MessageType_PREPARE &&
 		intendedForLeader &&
-		consensus.leaderSanityChecks(msg):
+		consensus.leaderSanityChecks(msg, senderKey):
 		consensus.onPrepare(msg)
 	case t == msg_pb.MessageType_COMMIT &&
 		intendedForLeader &&
-		consensus.leaderSanityChecks(msg):
+		consensus.leaderSanityChecks(msg, senderKey):
 		consensus.onCommit(msg)
-	case t == msg_pb.MessageType_VIEWCHANGE &&
-		consensus.viewChangeSanityCheck(msg):
```

Comment:
I think you still need `VerifyMessageSig` for view change messages right?

Reply:
right, added back.